app/controllers/hiring_staff/sessions_controller.rb

```diff
@@ -5,12 +5,20 @@ class HiringStaff::SessionsController < HiringStaff::BaseController
   skip_before_action :check_terms_and_conditions, only: %i[destroy]
 
   def destroy
-    session.destroy
-    redirect_to root_path, notice: I18n.t('messages.access.signed_out')
+    redirect_to logout_url
   end
 
   private
 
+  def logout_url
+    url = URI.parse(ENV['DFE_SIGN_IN_ISSUER'] + '/session/end')
+    url.query = {
+      post_logout_redirect_uri: auth_dfe_signout_url,
+      id_token_hint: session[:id_token]
+    }.to_query
```

Comment:
Could we move the `session.destroy` into this method? We want to be sure the session is destroyed, but we need to use some of the information in the session first to generate logout_url.

Reply:
I tested this idea. If you do this, then the method `destroy` on the other controller app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb never gets run, and signing out redirects you to the wrong page: /identifications/new.

So some part of the session must be necessary to do one or both of those. Session keys:
`["_csrf_token", "omniauth.nonce", "session_id", "urn", "multiple_schools", "id_token"]`